ydb/core/tx/schemeshard/schemeshard__operation_apply_build_index.cpp

```diff
@@ -50,24 +87,16 @@ TVector<ISubOperation::TPtr> ApplyBuildIndex(TOperationId nextId, const TTxTrans
     }
 
     if (!indexName.empty()) {
-        auto alterImplTableTransactionTemplate = [] (TPath index, TPath implIndexTable, TTableInfo::TPtr implIndexTableInfo) {
-            auto indexImplTableAltering = TransactionTemplate(index.PathString(), NKikimrSchemeOp::EOperationType::ESchemeOpFinalizeBuildIndexImplTable);
-            auto alterTable = indexImplTableAltering.MutableAlterTable();
-            alterTable->SetName(implIndexTable.LeafName());
-            alterTable->MutablePartitionConfig()->MutableCompactionPolicy()->CopyFrom(implIndexTableInfo->PartitionConfig().GetCompactionPolicy());
-            alterTable->MutablePartitionConfig()->MutableCompactionPolicy()->SetKeepEraseMarkers(false);
-            alterTable->MutablePartitionConfig()->SetShadowData(false);
-            return indexImplTableAltering;
-        };
-
         TPath index = table.Child(indexName);
-        for (const std::string_view implTable : NTableIndex::ImplTables) {
-            TPath implIndexTable = index.Child(implTable.data());
-            if (!implIndexTable.IsResolved()) {
-                continue;
+        Y_ABORT_UNLESS(index.Base()->GetChildren().size() >= 1);
+        for (auto& indexChildItems : index.Base()->GetChildren()) {
+            const auto& indexImplTableName = indexChildItems.first;
+            const auto partId = NextPartId(nextId, result);
+            if (indexImplTableName.EndsWith("tmp")) {
```

Comment:
I would prefer to remember or reconstruct actual temporary table names instead of guessing.
Is it possible at all?

Reply:
Fixed

Reply:
Yes, it's possible, but I think current approach is more flexible for potential new indexes.

Added some comments to explain this behavior